Foundation/String.swift

```diff
@@ -29,10 +29,13 @@ extension String : _ObjectiveCBridgeable {
             result = source._storage
         } else if type(of: source) == _NSCFString.self {
             let cf = unsafeBitCast(source, to: CFString.self)
+            let length = CFStringGetLength(cf)
             if let str = CFStringGetCStringPtr(cf, CFStringEncoding(kCFStringEncodingUTF8)) {
-                result = String(cString: str)
+                result = str.withMemoryRebound(to: UInt8.self, capacity: length) { ptr in
+                    let buffer = UnsafeBufferPointer(start: ptr, count: length)
+                    return String(decoding: buffer, as: UTF8.self)
+                }
```

Comment:
This looks good. If you're open to a small addition to your fix, we may want to also consider trying to call `CFStringGetCharactersPtr` if `CFStringGetCStringPtr` fails. That one may return a pointer to a UTF16 encoded buffer (or NULL), which we can also use without re-allocating in the slow path below.

Reply:
```
/* These functions attempt to return in O(1) time the desired format for the string.
Note that although this means a pointer to the internal structure is being returned,
this can't always be counted on. Please see note at the top of the file for more
details.
*/

CF_EXPORT
const UniChar *CFStringGetCharactersPtr(CFStringRef theString);  /* May return NULL at any time; be prepared for NULL, if not now, in some other time or place. See discussion at top of this file. */
```

Reply:
@pakera I will look into making that change. BTW, is it correct to use `CFStringGetLength` to find the number of bytes in the UTF8 string or is there something better to use?